[WebGPU] Parse literal integer/floats as AbstractInt/AbstractFloat

tuankiet65 opened the pull request:
#### 6559f40a2d93a6f9d8fb8f06f66dca5f3b49d642
<pre>
[WebGPU] Parse literal integer/floats as AbstractInt/AbstractFloat
<a href="https://bugs.webkit.org/show_bug.cgi?id=241755">https://bugs.webkit.org/show_bug.cgi?id=241755</a>
&lt;rdar://95925953&gt;

Reviewed by Geoffrey Garen and Myles C. Maxfield.

Literal integers and floats should be parsed as AbstractInt (int64_t)
and AbstractFloat (double). Current code parses them as int32_t and
float instead.

Tests: Source/WebGPU/WGSLUnitTests/Parser/ConstLiteralTests.mm

* Source/WebGPU/WGSL/AST/Expression.h:
(WGSL::AST::Expression::isAbstractIntLiteral const):
(WGSL::AST::Expression::isAbstractFloatLiteral const):
* Source/WebGPU/WGSL/AST/Expressions/LiteralExpressions.h:
* Source/WebGPU/WGSL/Parser.cpp:
(WGSL::Parser&lt;Lexer&gt;::parsePrimaryExpression):
* Source/WebGPU/WGSLUnitTests/Parser/ConstLiteralTests.mm: Added.
(parseLCharPrimaryExpression):
(-[ConstLiteralTests setUp]):
(-[ConstLiteralTests testBoolLiteralDecimal]):
(-[ConstLiteralTests testAbstractIntLiteralDecimal]):
(-[ConstLiteralTests testAbstractIntLiteralHex]):
(-[ConstLiteralTests testAbstractFloatLiteralDec]):
(-[ConstLiteralTests testAbstractFloatLiteralHex]):
* Source/WebGPU/WGSLUnitTests/WGSLParserTests.mm:
(-[WGSLParserTests testTrivialGraphicsShader]):
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/252060@main">https://commits.webkit.org/252060@main</a>
</pre>

geoffreygaren reviewed (approved):
r=me

On Slack I recommended changing override to final, and also a follow-up patch to change tokenizer type names to match class names for AbstractXLiteral.

tuankiet65 committed:
[WebGPU] Parse literal integer/floats as AbstractInt/AbstractFloat
https://bugs.webkit.org/show_bug.cgi?id=241755
<rdar://95925953>

Reviewed by Geoffrey Garen and Myles C. Maxfield.

Literal integers and floats should be parsed as AbstractInt (int64_t)
and AbstractFloat (double). Current code parses them as int32_t and
float instead.

Tests: Source/WebGPU/WGSLUnitTests/Parser/ConstLiteralTests.mm

* Source/WebGPU/WGSL/AST/Expression.h:
(WGSL::AST::Expression::isAbstractIntLiteral const):
(WGSL::AST::Expression::isAbstractFloatLiteral const):
* Source/WebGPU/WGSL/AST/Expressions/LiteralExpressions.h:
* Source/WebGPU/WGSL/Parser.cpp:
(WGSL::Parser<Lexer>::parsePrimaryExpression):
* Source/WebGPU/WGSLUnitTests/Parser/ConstLiteralTests.mm: Added.
(parseLCharPrimaryExpression):
(-[ConstLiteralTests setUp]):
(-[ConstLiteralTests testBoolLiteralDecimal]):
(-[ConstLiteralTests testAbstractIntLiteralDecimal]):
(-[ConstLiteralTests testAbstractIntLiteralHex]):
(-[ConstLiteralTests testAbstractFloatLiteralDec]):
(-[ConstLiteralTests testAbstractFloatLiteralHex]):
* Source/WebGPU/WGSLUnitTests/WGSLParserTests.mm:
(-[WGSLParserTests testTrivialGraphicsShader]):
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/252060@main
